### PR TITLE
fix issue when cuda arch is turned off

### PR DESCRIPTION
**Pull request by jeffmiles63**
Don't include arch setting in KOKKOS_CUDA_OPTIONS if the arch is turned off
Don't include arch setting in KOKKOS_LINK_OPTIONS at all.  nvcc_wrapper doesn't seem
to care, but clang complains about an invalid setting.

### CODE REVIEW

File: cmake/kokkos_arch.cmake [hunk 1]
```diff
@@ -389,11 +389,11 @@ IF(KOKKOS_ARCH_${ARCH})
     MESSAGE(FATAL_ERROR "Multiple GPU architectures given! Already have ${CUDA_ARCH_ALREADY_SPECIFIED}, but trying to add ${ARCH}. If you are re-running CMake, try clearing the cache and running again.")
   ENDIF()
   SET(CUDA_ARCH_ALREADY_SPECIFIED ${ARCH} PARENT_SCOPE)
-  GLOBAL_APPEND(KOKKOS_CUDA_OPTIONS "${CUDA_ARCH_FLAG}=${FLAG}")
-  GLOBAL_APPEND(KOKKOS_LINK_OPTIONS "${CUDA_ARCH_FLAG}=${FLAG}")
```
Comment (masterleinad) [inline cmake/kokkos_arch.cmake]: We need this for `KOKKOS_ENABLE_RELOCATABLE_DEVICE_CODE=ON`, see #2400.

**Comment by jeffmiles63** [comment]
@crtrott @jjwilke @dalg24 
There are not currently any nightly test in the matrix that use Clang with RDC.  Is this something we should support or can we just say no RDC if using Clang?

**Comment by masterleinad** [comment]
> There are not currently any nightly test in the matrix that use Clang with RDC. Is this something we should support or can we just say no RDC if using Clang?

This does currently not work anyway. Let's just open a separate issue and move forward here. I don't think we need extra syntax for `clang` for the current issue if we just add the `-sm=arch_*` for relocatable device code.

File: cmake/kokkos_arch.cmake (hunk 1 above)
Comment (jeffmiles63) [reply cmake/kokkos_arch.cmake]: @dalg24 @jjwilke @crtrott @masterleinad 
Add link options back in for relocatable device code enabled.  We will sort out the Clang linking problem with a different issue.  This shouldn't affect any existing tests.

**Pull request merged**